### PR TITLE
Remove subsystem limit.

### PR DESCRIPTION
**Pull request by MageKing17**
Allows support for as many subsystems as available memory allows by replacing the `Ship_subsystems[]` pointer array with a pointer vector. It still just contains pointers to dynamically-allocated subsystem arrays (allocated in blocks of `NUM_SHIP_SUBSYSTEMS_PER_SET`), so this should behave almost identically to the current code.

Used a test build to verify that a mission that was exceeding the old subsystem limit generated no such errors with these changes.

### CODE REVIEW

File: code/ship/ship.cpp [hunk 1]
```diff
@@ -5165,9 +5162,8 @@ void ship_init()
 			ships_inited = 1;
 		}
 
-		// NULL out "dynamic" subsystem ptr's
-		for (i = 0; i < NUM_SHIP_SUBSYSTEM_SETS; i++)
-			Ship_subsystems[i] = NULL;
+		// We shouldn't already have any subsystem pointers at this point.
+		Assertion(Ship_subsystems.empty(), "Some pre-allocated subsystems didn't get cleared out: %d batches present during ship_init(); get a coder!\n", Ship_subsystems.size());
```
Comment (asarium) [inline code/ship/ship.cpp]: std::vector::size() returns a size_t but the format string expects an int. You can use the `SIZE_T_ARG` macro for cross-platform `size_t` formatting.

Comment (MageKing17) [inline code/ship/ship.cpp]: Oh right; I'll fix that when I'm back at my desktop.

**Comment by asarium** [comment]
LGTM. This can be merged when the CI checks have passed.

**Comment by chief1983** [comment]
Not likely something like this would affect multiplayer adversely?

**Comment by The-E** [comment]
I can't see how.

**Pull request merged**